### ISSUE TITLE
Mobile/Move IsMoving check is not 100% reliable while turning

### ISSUE THREAD

**Issue by reaperrr** [OpenRA/OpenRA#11885]
According to my tests with `WithMoveAnimation` in conjunction with `Mobile` ground actors, the `IsMoving` check occasionally seems to return `false` while turning, despite the recent attempts to improve the `IMove.IsMoving` reliability.

**Comment by GraionDilach** [comment]
Related to #11895 I guess.

**Comment by reaperrr** [comment]
I was able to narrow this down, I hope we can get this fixed for Next+1.

Turns out this is at least related to #13346, afterall.
By building a test-case that displays the current ChildActivity as FloatingText, I was able to see that, at the crucial moment when this 'stoppage' happens, the actor runs Turn as child activity for exactly 2 ticks. Increasing TurnSpeed appears to have no effect on that.

For the majority of actors, however, we only want the Turn activity to count as movement in this particular situation, not in general, so simply making Turn set IsMoving to 'true' won't fix this properly (it will open up regressions elsewhere).
Also, since this stops (forward-)movement as well, we might rather try to find a way to avoid these invisible mini-turns (I say invisible because even 32 facings are not enough to visually see this turn). Maybe just force-set the facing to the desired facing instead of a "proper" turn if the delta is less than a certain value?
